apps/oxlint/src/command/lint.rs

```diff
@@ -211,64 +211,178 @@ impl FromStr for OutputFormat {
-    #[bpaf(switch, hide_usage)]
-    pub jsdoc_plugin: bool,
+    #[bpaf(flag(Enable::Enable, Enable::NotSet), hide_usage)]
+    pub jsdoc_plugin: Enable,
 
     /// Enable the Jest plugin and detect test problems
-    #[bpaf(switch, hide_usage)]
-    pub jest_plugin: bool,
+    #[bpaf(flag(Enable::Enable, Enable::NotSet), hide_usage)]
+    pub jest_plugin: Enable,
 
     /// Enable the Vitest plugin and detect test problems
-    #[bpaf(switch, hide_usage)]
-    pub vitest_plugin: bool,
+    #[bpaf(flag(Enable::Enable, Enable::NotSet), hide_usage)]
+    pub vitest_plugin: Enable,
 
     /// Enable the JSX-a11y plugin and detect accessibility problems
-    #[bpaf(switch, hide_usage)]
-    pub jsx_a11y_plugin: bool,
+    #[bpaf(flag(Enable::Enable, Enable::NotSet), hide_usage)]
+    pub jsx_a11y_plugin: Enable,
 
     /// Enable the Next.js plugin and detect Next.js problems
-    #[bpaf(switch, hide_usage)]
-    pub nextjs_plugin: bool,
+    #[bpaf(flag(Enable::Enable, Enable::NotSet), hide_usage)]
+    pub nextjs_plugin: Enable,
 
     /// Enable the React performance plugin and detect rendering performance problems
-    #[bpaf(switch, hide_usage)]
-    pub react_perf_plugin: bool,
+    #[bpaf(flag(Enable::Enable, Enable::NotSet), hide_usage)]
+    pub react_perf_plugin: Enable,
 
     /// Enable the promise plugin and detect promise usage problems
-    #[bpaf(switch, hide_usage)]
-    pub promise_plugin: bool,
+    #[bpaf(flag(Enable::Enable, Enable::NotSet), hide_usage)]
+    pub promise_plugin: Enable,
 
     /// Enable the node plugin and detect node usage problems
-    #[bpaf(switch, hide_usage)]
-    pub node_plugin: bool,
+    #[bpaf(flag(Enable::Enable, Enable::NotSet), hide_usage)]
+    pub node_plugin: Enable,
 
     /// Enable the security plugin and detect security problems
-    #[bpaf(switch, hide_usage)]
-    pub security_plugin: bool,
+    #[bpaf(flag(Enable::Enable, Enable::NotSet), hide_usage)]
+    pub security_plugin: Enable,
+}
+
+#[derive(Default, Debug, Clone, Copy, PartialEq, Eq)]
+#[allow(clippy::enum_variant_names)]
+pub enum Enable {
+    Enable,
+    Disable,
+    #[default]
+    NotSet,
+}
+
+impl From<Option<bool>> for Enable {
+    fn from(value: Option<bool>) -> Self {
+        match value {
+            Some(true) => Self::Enable,
+            Some(false) => Self::Disable,
+            None => Self::NotSet,
+        }
+    }
+}
+
+impl From<Enable> for Option<bool> {
+    fn from(value: Enable) -> Self {
+        match value {
+            Enable::Enable => Some(true),
+            Enable::Disable => Some(false),
+            Enable::NotSet => None,
+        }
+    }
+}
+
+impl Enable {
+    #[inline]
+    pub fn is_enabled(self) -> bool {
+        matches!(self, Self::Enable)
+    }
+
+    #[inline]
+    pub fn is_not_set(self) -> bool {
+        matches!(self, Self::NotSet)
+    }
+
+    pub fn inspect<F>(self, f: F)
+    where
+        F: FnOnce(bool),
+    {
```

Comment:
should this return something? to be consistent w/ Rust I guess it should return `Enable`?

Reply:
No. It's based of `Option::inspect`.

Reply:
it looks like `Option::inspect` returns `Self` though: https://doc.rust-lang.org/1.81.0/src/core/option.rs.html#1135

```rust
pub fn inspect<F: FnOnce(&T)>(self, f: F) -> Self {}
```

not really a big deal, feel free to resolve if intentional